src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+        this(Storage.DEFAULT_PATH, gui);
+    }
+
+    /**
+     * constructor to specify file path of the last saved data
+     *
+     * @param filePath = path of last saved data file
+     * @param gui      = using gui?
+     */
+    public Duke(String filePath, boolean gui) {
+        this.ui = new Ui(new Scanner(System.in));
+        this.storage = new Storage(filePath);
+        if (this.storage.fileExist()) {
+            try {
+                this.tasks = TaskList.fromCSVList(storage.loadCSVList());
+            } catch (Exception e) {
+                this.ui.respond(Ui.loadingErrorMsg);
+                tasks = new TaskList();
+            }
+        } else {
+            tasks = new TaskList();
+        }
+        /*
+        if (gui) {
+            this.gui = new Gui(this.dialogContainer, this.userInput);
+        }*/
+    }
+
+    /**
+     * Main method.
+     */
+    public void run() {
+        ui.respond(Ui.greetings);
+        Command cmd;
+        while (ui.hasNextLine()) {
+            cmd = Parser.parse(ui.nextLine());
+            cmd.execute(this.tasks, this.ui, this.storage);
+        }
+    }
+
+    public static void main(String[] args) {
+        if (args.length > 0) {
+            new Duke(args[0], false).run();
+        } else {
+            new Duke(false).run();
+        }
+    }
+
+    @FXML
+    protected void handleSendButtonAction(MouseEvent event) {
+        try {
+            this.ui = new Gui(this.dialogContainer, this.userInput);
+            if(this.ui.hasNextLine()) {
+                Command cmd = Parser.parse(this.ui.nextLine());
+                cmd.execute(this.tasks, this.ui = new Gui(this.dialogContainer, this.userInput), this.storage);
+                this.ui.clearUserInput();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @FXML
+    protected void handleUserInputOnAction(Event event) {
+        this.ui = new Gui(this.dialogContainer, this.userInput);
+        if(this.ui.hasNextLine()) {
+            Command cmd = Parser.parse(this.ui.nextLine());
+            cmd.execute(this.tasks, this.ui = new Gui(this.dialogContainer, this.userInput), this.storage);
+            this.ui.clearUserInput();
+        }
+    }
+
+    @Override
+    public void start(Stage stage) throws Exception {
+        URL url = new File("src/main/java/duke/fxml/main.fxml").toURI().toURL();
+        Parent root = FXMLLoader.load(url);
+        //Parent root = FXMLLoader.load(getClass().getClassLoader().getResource("src/main/java/duke/fxml/main.fxml"));
+
+        Scene scene = new Scene(root, 300, 275);
+
+        stage.setTitle("FXML Welcome");
+        stage.setScene(scene);
+        stage.show();
+
+        stage.setTitle("Duke");
+        stage.setResizable(false);
+        stage.setMinHeight(600.0);
+        stage.setMinWidth(400.0);
+
+        //this.dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+    }
+/*
+    @Override
```

Comment:
Same here. If you do not require a code block please remove it instead of commenting. The code will be always available in the previous commit so that if you want you can check.